apps/remix-ide/src/app/panels/file-panel.js

```diff
@@ -103,194 +76,51 @@ module.exports = class Filepanel extends ViewPlugin {
    * @param callback (...args) => void
    */
   registerContextMenuItem (item) {
-    if (!item) throw new Error('Invalid register context menu argument')
-    if (!item.name || !item.id) throw new Error('Item name and id is mandatory')
-    if (!item.type && !item.path && !item.extension && !item.pattern) throw new Error('Invalid file matching criteria provided')
-    if (this.registeredMenuItems.filter((o) => {
-      return o.id === item.id && o.name === item.name
-    }).length) throw new Error(`Action ${item.name} already exists on ${item.id}`)
-    this.registeredMenuItems = [...this.registeredMenuItems, item]
-    this.removedMenuItems = this.removedMenuItems.filter(menuItem => item.id !== menuItem.id)
-    this.renderComponent()
+    this.emit('registerContextMenuItem', item)
   }
 
   removePluginActions (plugin) {
-    this.registeredMenuItems = this.registeredMenuItems.filter((item) => {
-      if (item.id !== plugin.name || item.sticky === true) return true
-      else {
-        this.removedMenuItems.push(item)
-        return false
-      }
-    })
-    this.renderComponent()
+    this.emit('removePluginActions', plugin)
   }
 
-  async getCurrentWorkspace () {
-    return await this.request.getCurrentWorkspace()
+  getCurrentWorkspace () {
+    return this.workspaceStatus
```

Comment:
is `workspaceStatus` defining the current workspace? maybe the wording can be improved